# Исследование надежности заемщиков

**Описание проекта**

Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.
Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

**Описание данных**

- children — количество детей в семье
- days_employed — общий трудовой стаж в днях
- dob_years — возраст клиента в годах
- education — уровень образования клиента
- education_id — идентификатор уровня образования
- family_status — семейное положение
- family_status_id — идентификатор семейного положения
- gender — пол клиента
- income_type — тип занятости
- debt — имел ли задолженность по возврату кредитов
- total_income — ежемесячный доход
- purpose — цель получения кредита

Предобработка данных
- проверка на заполненность
- проверка на аномалии
- проверка
Исследуйте данные и ответьте на вопросы

In [1]:
#Импортируем библиотеку pandas
import pandas as pd

In [2]:
#Откроем файл
try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

In [3]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

In [4]:
#Выведем основную информацию о датафрейме с помощью метода `info()`.
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

In [5]:
#Выведем количество пропущенных значений для каждого столбца.
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце мы обработаем на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.

In [6]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. Таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. 

In [7]:
data['days_employed'] = data['days_employed'].abs()

Для каждого типа занятости выведем медианное значение трудового стажа `days_employed` в днях.

In [8]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставим их как есть.

In [9]:
#Выведем перечень уникальных значений столбца `children`
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

В столбце `children` есть два аномальных значения. Удалим строки, в которых встречаются такие аномальные значения из датафрейма `data`.

In [10]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

Ещё раз выведем перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.

In [11]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

### Удаление пропусков (продолжение)

Заполним пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.

In [12]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

In [13]:
#Убедимся, что все пропуски заполнены. Проверим себя и ещё раз выведем количество пропущенных значений для каждого столбца.
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

In [14]:
#Заменим вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

Обработаем неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведем их к нижнему регистру.

In [15]:
data['education'] = data['education'].str.lower()

Выведем на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалим их.

In [16]:
data.duplicated().sum()

71

In [17]:
data = data.drop_duplicates()

### Категоризация данных

На основании диапазонов, указанных ниже, создадим в датафрейме `data` столбец `total_income_category` с категориями:

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.

In [18]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [19]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

In [20]:
#Выведим на экран перечень уникальных целей взятия кредита из столбца `purpose`.
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

Создадим функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.

In [21]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [22]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Исследование данных

#### Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [23]:
#для формулировки вывода мы можем основываться на следующие столбцы таблицы data: 
#количество детей и имел ли задолженность по возврату кредитов 
#выведем данные о количестве клиентов у которых есть задолженность и нет (1/0 соответственно)
data.groupby('debt')['children'].count()

debt
0    19599
1     1732
Name: children, dtype: int64

In [24]:
#сгруппируем клиентов по количеству детей
data.groupby('children')['debt'].count()

children
0    14091
1     4808
2     2052
3      330
4       41
5        9
Name: debt, dtype: int64

Проведем категоризацию данных. Для повышения репрезентативности объединим данные по клиентам с 2, 3, 4, 5. В данном случае по отношению ко всему датасету сумма данных будет составлять 11-12%.

In [25]:
#Для категоризации данных используем функцию
def categorize_child(children):
    try:
        if children == 0:
            return 'нет детей'
        elif children == 1:
            return '1 ребенок'
        elif children >= 2:
            return '2 и более детей'
    except:
        pass

In [26]:
data['categorize_child'] = data['children'].apply(categorize_child)

In [27]:
#Выведем сводную таблицу с данными о количестве детей - соответствующим индексу таблицы, 
#и сумма клиентов имеющих задолженность или нет в значениях
data_pivot = data.pivot_table(index=['categorize_child'], columns='debt', values='days_employed', aggfunc='count')
display(data_pivot) 

debt                  0     1
categorize_child             
1 ребенок          4364   444
2 и более детей    2207   225
нет детей         13028  1063

In [28]:
#Выведем информацию по таблице.
data_pivot.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3 entries, 1 ребенок to нет детей
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   0       3 non-null      int64
 1   1       3 non-null      int64
dtypes: int64(2)
memory usage: 72.0+ bytes


In [29]:
#Составим сводную таблицу с клиентами по наличию задолженности
data_pivot = data_pivot.rename(columns={0: 'no_debts', 1: 'have_debt'})
display(data_pivot)

debt              no_debts  have_debt
categorize_child                     
1 ребенок             4364        444
2 и более детей       2207        225
нет детей            13028       1063

In [30]:
#найдем отношение числа должников к общему числу клиентов категории
data_pivot['debt_percent'] = data_pivot['have_debt'] / (data_pivot['no_debts'] + data_pivot['have_debt'])
#Перезапишем индекс в таблице data_pivot.
data_pivot = data_pivot.sort_values(by='debt_percent').reset_index()
display(data_pivot)

debt categorize_child  no_debts  have_debt  debt_percent
0           нет детей     13028       1063      0.075438
1           1 ребенок      4364        444      0.092346
2     2 и более детей      2207        225      0.092516

In [31]:
#выведем полученные данные в текстовой формате
for i in data_pivot.index:
    pivot = data_pivot.loc[i, 'debt_percent']
    child = data_pivot.loc[i, 'categorize_child']
    print(f'У клиентов с количеством детей = {child} \n вероятность не возврата кредита = {pivot:.2%}')

У клиентов с количеством детей = нет детей 
 вероятность не возврата кредита = 7.54%
У клиентов с количеством детей = 1 ребенок 
 вероятность не возврата кредита = 9.23%
У клиентов с количеством детей = 2 и более детей 
 вероятность не возврата кредита = 9.25%


**Вывод:** 

По полученным данным категоризированной таблицы можно сказать, что не платежеспособность по кредитам увеличивается прямо пропорционально количеству детей. Вероятными причинами могут быть повышенная финансовая нагрузка на бюджет семьи. 
Наибольший процент заемщиков с долгом - клиенты с 2 детьми (9,25%)
Наименьший процент заемщиков с долгом - клиенты без детей (7,54%)

#### Есть ли зависимость между семейным положением и возвратом кредита в срок?

На основе методов, реализованных ранее проведем анализ зависимости семейного положения и возвратом кредита в срок.

In [32]:
#выведем все возможные статусы, прадставленные в таблице
data.value_counts('family_status')

family_status
женат / замужем          12261
гражданский брак          4134
Не женат / не замужем     2796
в разводе                 1189
вдовец / вдова             951
dtype: int64

In [33]:
#Выведем сводную таблицу с данными о семейном положении и суммой количества клиентов имеющих задолженность или нет
data_status = data.pivot_table(index=['family_status'], columns='debt', values='days_employed', aggfunc='count')
data_status = data_status.rename(columns={0: 'no_debts', 1: 'have_debt'})
display(data_status) 

debt                   no_debts  have_debt
family_status                             
Не женат / не замужем      2523        273
в разводе                  1105         84
вдовец / вдова              888         63
гражданский брак           3749        385
женат / замужем           11334        927

In [34]:
#найдем отношение числа должников к общему числу клиентов категории
#также для удобства работы перезапишм индексы в таблице
data_status['debt_percent'] = data_status['have_debt'] / (data_status['no_debts'] + data_status['have_debt'])
data_status = data_status.sort_values(by='debt_percent').reset_index()
display(data_status)

debt          family_status  no_debts  have_debt  debt_percent
0            вдовец / вдова       888         63      0.066246
1                 в разводе      1105         84      0.070648
2           женат / замужем     11334        927      0.075606
3          гражданский брак      3749        385      0.093130
4     Не женат / не замужем      2523        273      0.097639

In [35]:
#выведем полученные данные в текстовой формате
for i in data_status.index:
    pivot = data_status.loc[i, 'debt_percent']
    stat = data_status.loc[i, 'family_status']
    print(f'У клиентов в статусе = {stat} \n вероятность задолженности = {pivot:.2%}')

У клиентов в статусе = вдовец / вдова 
 вероятность задолженности = 6.62%
У клиентов в статусе = в разводе 
 вероятность задолженности = 7.06%
У клиентов в статусе = женат / замужем 
 вероятность задолженности = 7.56%
У клиентов в статусе = гражданский брак 
 вероятность задолженности = 9.31%
У клиентов в статусе = Не женат / не замужем 
 вероятность задолженности = 9.76%


**Вывод:** 

По полученным данным видно, что вероятность не вернуть кредит выше у категорй людей в статусе "гражданский брак" и "Не женат / не замужем" составляет 9-10%. Наименьший процент задолженностей у клиентов в статусе "вдовец / вдова" и "в разводе" - 6-7%.
Возможными причинами поличеного результата для перечисленных категорий клиентов могут быть: небольшой заработок, возраст, квалифицированность. 

#### Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [36]:
#выведем все возможные статусы, прадставленные в таблице
data.value_counts('total_income_category')

total_income_category
C    15921
B     5014
D      349
A       25
E       22
dtype: int64

In [37]:
#Выведем сводную таблицу с данными о категорях уровня доходов и суммой количества клиентов имеющих задолженность или нет
#для удобства добавим столбец range с расшифровкой дохода по категории
data_income = data.pivot_table(index=['total_income_category'], columns='debt', values='days_employed', aggfunc='count')
data_income = data_income.rename(columns={0: 'no_debts', 1: 'have_debt'})
data_income['range'] = ['1000001 и выше', '200001–1000000', '50001–200000', '30001–50000', '0–30000']
display(data_income) 

debt                   no_debts  have_debt           range
total_income_category                                     
A                            23          2  1000001 и выше
B                          4660        354  200001–1000000
C                         14568       1353    50001–200000
D                           328         21     30001–50000
E                            20          2         0–30000

Выборка по клиентам категой А и Е мала для точного определения зависимости, так как доходы находящиеся в этих промежутках являются "начальными" и "конечными" для совокупности, также столбец D содержит небольшое количество значений, поэтому категории не являются показательными. 

In [38]:
#найдем отношение числа должников к общему числу клиентов категории
#также для удобства работы перезапишм индексы в таблице
data_income['debt_percent'] = data_income['have_debt'] / (data_income['no_debts'] + data_income['have_debt'])
data_income = data_income.sort_values(by='debt_percent').reset_index()
display(data_income)

debt total_income_category  no_debts  have_debt           range  debt_percent
0                        D       328         21     30001–50000      0.060172
1                        B      4660        354  200001–1000000      0.070602
2                        A        23          2  1000001 и выше      0.080000
3                        C     14568       1353    50001–200000      0.084982
4                        E        20          2         0–30000      0.090909

In [39]:
#выведем полученные данные в текстовой формате
for i in data_income.index:
    pivot = data_income.loc[i, 'debt_percent']
    income = data_income.loc[i, 'range']
    print(f'У клиентов с доходом в диапазоне = {income} \n вероятность задолженности = {pivot:.2%}')

У клиентов с доходом в диапазоне = 30001–50000 
 вероятность задолженности = 6.02%
У клиентов с доходом в диапазоне = 200001–1000000 
 вероятность задолженности = 7.06%
У клиентов с доходом в диапазоне = 1000001 и выше 
 вероятность задолженности = 8.00%
У клиентов с доходом в диапазоне = 50001–200000 
 вероятность задолженности = 8.50%
У клиентов с доходом в диапазоне = 0–30000 
 вероятность задолженности = 9.09%


**Вывод:** 

Из ранее определенных показательных значений - категорий В и С - с процентным отношением должников к общему количеству клиентов категории = 7-8.5%, можно сделать вывод, что клиенты с большим доходом чаще возвращают кредит.

#### Как разные цели кредита влияют на его возврат в срок?

In [40]:
#выведем все возможные категории, прадставленные в таблице
data.value_counts('purpose_category')

purpose_category
операции с недвижимостью    10751
операции с автомобилем       4279
получение образования        3988
проведение свадьбы           2313
dtype: int64

In [41]:
#Выведем сводную таблицу с данными о целях кредита и суммой количества клиентов имеющих задолженность или нет
#для удобства добавим столбец range с расшифровкой дохода по категории
data_purpose = data.pivot_table(index=['purpose_category'], columns='debt', values='days_employed', aggfunc='count')
data_purpose = data_purpose.rename(columns={0: 'no_debts', 1: 'have_debt'})
display(data_purpose)

debt                      no_debts  have_debt
purpose_category                             
операции с автомобилем        3879        400
операции с недвижимостью      9971        780
получение образования         3619        369
проведение свадьбы            2130        183

In [42]:
#найдем отношение числа должников к общему числу клиентов категории
#также для удобства работы перезапишм индексы в таблице
data_purpose['debt_percent'] = data_purpose['have_debt'] / (data_purpose['no_debts'] + data_purpose['have_debt'])
data_purpose = data_purpose.sort_values(by='debt_percent').reset_index()
display(data_purpose)

debt          purpose_category  no_debts  have_debt  debt_percent
0     операции с недвижимостью      9971        780      0.072551
1           проведение свадьбы      2130        183      0.079118
2        получение образования      3619        369      0.092528
3       операции с автомобилем      3879        400      0.093480

In [43]:
#выведем полученные данные в текстовой формате
for i in data_purpose.index:
    pivot = data_purpose.loc[i, 'debt_percent']
    purpose = data_purpose.loc[i, 'purpose_category']
    print(f'У клиентов, цель которых {purpose} \n вероятность задолженности = {pivot:.2%}')

У клиентов, цель которых операции с недвижимостью 
 вероятность задолженности = 7.26%
У клиентов, цель которых проведение свадьбы 
 вероятность задолженности = 7.91%
У клиентов, цель которых получение образования 
 вероятность задолженности = 9.25%
У клиентов, цель которых операции с автомобилем 
 вероятность задолженности = 9.35%


**Вывод:** 

Клиенты взявшие кредит под цели "операции с недвижимостью" и "проведение свадьбы" чаще отдают кредит в срок. Возможной причиной может быть получение финансовой выгоды в процессе достижения цели, также срок на который предоставлен кредит. 

#### Возможные причины появления пропусков в исходных данных.

1) человеческий фактор - пользователь забыл или опечатался
2) отсутствие значения - вместо 0 или другой переменной поле остается пустым
3) отсутствие в нужных данных в списке предложенных
4) сбой программ
5) неверное преобразование формата данных

#### Почему заполнить пропуски медианным значением — лучшее решение для количественных переменных?

Если в количественных переменных присутствует пропуск, но при этом остальные графы несут информационную ценность, то есть строка не может быть удалена, пропуск необходимо заполнить значением, для проведения дальнейших операций с таблицей. 
Выборка данных может быть разного объема, поэтому отсутствие одного количественного значения, например, в небольшой таблице, может значительно исказить анализируемые данные. Поэтому для поддержания актуальности данных, следует заменить переменную средним(медианным) значением, чтобы данная строка не выбивалась из всей совокупности и не являлась аномальной.

### Общий вывод.

В ходе проведенной работы были проанализированы данные платежеспособности заемщиков. Была проведена предобработка данных: проверка пустых значений, поиск анамалий, преобразование данных.
На основе полученных данных были рассмотрены следующие вопросы:

1) Зависимость между количеством детей и возвратом кредита в срок

Данные по категории с минимальной вероятностью - 7.54% для клиентов у ктороых нет детей.
Наивысший процент возможной задолженности - 9.25% для категории клиентов с двумя и более детьми.
По полученным данным категоризированной таблицы можно сказать, что не платежеспособность по кредитам увеличивается прямо пропорционально количеству детей. Вероятными причинами могут быть повышенная финансовая нагрузка на бюджет семьи.

2) Зависимость между семейным положением и возвратом кредита в срок

Наименьший процент задолженностей наблюдается у людей в статусе "вдовец / вдова" - 6.62%, "в разводе " - 7.06%
Наибольший процент в статусе "гражданский брак"- 9.31% и "Не женат / не замужем" - 9.76%
Возможными причинами данной тенденции может являться возраст заемщиков, их квалификация и доход.

3) Зависимость между уровнем дохода и возвратом кредита в срок

В данной выборке данные большей части категорий доходов нельзя назвать показательными. Так как наблюдается сильная разница в значениях количества наблюдений. Для более точной информации необходимо рассматривать выборку с более детальньной категоризацией данных.

4) Как разные цели кредита влияют на его возврат в срок

Наименьший процент задолженностей наблюдается у клиентов, взявших кредит под цели "операции с недвижимостью" и "проведение свадьбы".
Наибольший процент для целей "получение образования" и "операции с автомобилем".
Возможной причиной наличия задолженности может быть: срок, на который предоставлен кредит, также возможное получение финансовой выгоды в процессе достижения цели.

Отклонение на несколько процентов, указанных в сводных таблицых является колоссальным в пределах бюджета страны. Поэтому данными значениями и их отклонениями пренебрегать в процессе анализа не стоит.

Из полученых результатов можно заключить, что на возврат заемщиком кредита в срок, влияют в разной степени разнообразные аспекты от его дохода до семейного положения. Для выявления более точных возможных причин необходимо проводить анализ требуемых величин.